File: temp/fix_encoding.py
import os

files_to_fix = [
    r'src\app\(admin)\admin\page.tsx',
    r'src\app\(admin)\admin\settings\page.tsx',
    r'src\app\(auth)\auth\forgot-password\ForgotPasswordClient.tsx',
    r'src\app\(auth)\auth\steps\EmailStep.tsx',
    r'src\app\(auth)\auth\steps\EnterStep.tsx',
    r'src\app\(storefront)\cart\CartClient.tsx',
    r'src\app\(storefront)\order\failure\OrderFailureClient.tsx',
    r'src\components\admin\products\ProductForm.tsx',
    r'src\components\ErrorBoundary.tsx',
    r'src\components\PhoneVerificationModal.tsx',
]

for file_path in files_to_fix:
    try:
        # Read file
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Remove BOM and fix HTML entities
        content = content.replace('\ufeff', '')
        content = content.replace('&apos;', "'")
        content = content.replace('&quot;', '"')
        content = content.replace('&amp;', '&')
        
        # Write back
        with open(file_path, 'w', encoding='utf-8', newline='\r\n') as f:
            f.write(content)
        
        print(f"Fixed: {file_path}")
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")

print("\nAll files processed!")

File: temp/fix-encoding.js
const fs = require('fs');
const path = require('path');

const filesToFix = [
  'src\\app\\(admin)\\admin\\page.tsx',
  'src\\app\\(admin)\\admin\\settings\\page.tsx',
  'src\\app\\(auth)\\auth\\forgot-password\\ForgotPasswordClient.tsx',
  'src\\app\\(auth)\\auth\\steps\\EmailStep.tsx',
  'src\\app\\(auth)\\auth\\steps\\EnterStep.tsx',
  'src\\app\\(storefront)\\cart\\CartClient.tsx',
  'src\\app\\(storefront)\\order\\failure\\OrderFailureClient.tsx',
  'src\\components\\admin\\products\\ProductForm.tsx',
  'src\\components\\ErrorBoundary.tsx',
  'src\\components\\PhoneVerificationModal.tsx'
];

filesToFix.forEach(filePath => {
  try {
    // Read file
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Remove BOM and fix HTML entities
    content = content.replace(/^\uFEFF/, '');
    content = content.replace(/&apos;/g, "'");
    content = content.replace(/&quot;/g, '"');
    content = content.replace(/&amp;/g, '&');
    
    // Write back
    fs.writeFileSync(filePath, content, 'utf8');
    
    console.log(`Fixed: ${filePath}`);
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
  }
});

console.log('\nAll files processed!');
